src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -555,7 +555,21 @@ public void AddLowercase(CultureInfo culture)
                     }
                     else
                     {
-                        AddLowercaseRange(range.First, range.Last);
+                        char lower = culture.TextInfo.ToLower(range.First);
```

Comment:
As long as we're changing this code, might be worth caching culture.TextInfo into a local and then just using that local repeatedly throughout the method, since it's a virtual call.